src/dependencies.ts

```diff
@@ -0,0 +1,90 @@
+import config from './config';
+import {
+    connect,
+    getAndSetupDatabase,
+    COLLECTION_SOURCES,
+    COLLECTION_MAPPINGS,
+    COLLECTION_TARGETS,
+    COLLECTION_RESPONSES
+} from './database';
+import { buildSourcesService, SourcesService } from './services/sources';
+import { buildMappingsService, MappingsService } from './services/mappings';
+import { buildTargetsService, TargetsService } from './services/targets';
+import { buildResponsesService, ResponsesService } from './services/responses';
+import { buildMapperService, MapperService } from './services/mapper';
+import { buildRequesterService, RequesterService } from './services/requester';
+import { buildHttpEngineService, HttpEngineService } from './services/http-engine';
+import type { MongoClient, Db, Collection } from 'mongodb';
+
+export type DependenciesConfig = {
+    DBNAME?: string;
+    SOURCES?: string;
+    MAPPINGS?: string;
+    TARGETS?: string;
+    RESPONSES?: string;
+};
+export type DependenciesName = 'config'|'dbClient'|'db'|'sourcesCollection'|'sourcesService'|'mappingsCollection'|'mappingsService'|'targetsCollection'|'targetsService'
```

Comment:
I meant this could be
`export type DependenciesName = keyof Dependencies;`

Reply:
Great!